math/moserdebruijnsequence/sequence_test.go

```diff
@@ -0,0 +1,28 @@
+package moserdebruijnsequence
+
+import (
+	"reflect"
+	"testing"
+)
+
+var testCases = []struct {
+	name     string
+	inputNum int
+	expected []int
+}{
+	{"first 0 terms", 0, []int{}},
+	{"first 5 terms", 5, []int{0, 1, 4, 5, 16}},
+	{"first 10 terms", 10, []int{0, 1, 4, 5, 16, 17, 20, 21, 64, 65}},
```

Comment:
No need to test the first 0, 5, and 10 terms if you're testing the first 15 terms too